apis/kubedb/v1alpha2/mssql_webhook.go

```diff
@@ -0,0 +1,287 @@
+		return
+	}
+	mssqllog.Info("default", "name", m.Name)
+
+	m.SetDefaults()
+}
+
+//+kubebuilder:webhook:path=/validate-kubedb-com-v1alpha2-mssql,mutating=false,failurePolicy=fail,sideEffects=None,groups=kubedb.com,resources=mssqls,verbs=create;update,versions=v1alpha2,name=vmssql.kb.io,admissionReviewVersions=v1
+
+var _ webhook.Validator = &MSSQL{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (m *MSSQL) ValidateCreate() (admission.Warnings, error) {
+	mssqllog.Info("validate create", "name", m.Name)
+
+	allErr := m.ValidateCreateOrUpdate()
+	if len(allErr) == 0 {
+		return nil, nil
+	}
+	return nil, apierrors.NewInvalid(schema.GroupKind{Group: "kubedb.com", Kind: "MSSQL"}, m.Name, allErr)
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
+func (m *MSSQL) ValidateUpdate(old runtime.Object) (admission.Warnings, error) {
+	mssqllog.Info("validate update", "name", m.Name)
+
+	oldMSSQL := old.(*MSSQL)
+	allErr := m.ValidateCreateOrUpdate()
+
+	if m.Spec.Topology == nil && ptr.Deref(oldMSSQL.Spec.Replicas, 0) == 1 && ptr.Deref(m.Spec.Replicas, 0) > 1 {
+		allErr = append(allErr, field.Invalid(field.NewPath("spec").Child("replicas"),
+			m.Name,
+			"Cannot scale up from 1 to more than 1 in standalone mode"))
+	}
+
+	if len(allErr) == 0 {
+		return nil, nil
+	}
+
+	return nil, apierrors.NewInvalid(schema.GroupKind{Group: "kubedb.com", Kind: "MSSQL"}, m.Name, allErr)
+}
+
+// ValidateDelete implements webhook.Validator so a webhook will be registered for the type
+func (m *MSSQL) ValidateDelete() (admission.Warnings, error) {
+	mssqllog.Info("validate delete", "name", m.Name)
+
+	var allErr field.ErrorList
+	if m.Spec.TerminationPolicy == TerminationPolicyDoNotTerminate {
+		allErr = append(allErr, field.Invalid(field.NewPath("spec").Child("terminationPolicy"),
+			m.Name,
+			"Can not delete as terminationPolicy is set to \"DoNotTerminate\""))
+		return nil, apierrors.NewInvalid(schema.GroupKind{Group: "kubedb.com", Kind: "MSSQL"}, m.Name, allErr)
+	}
+	return nil, nil
+}
+
+func (m *MSSQL) ValidateCreateOrUpdate() field.ErrorList {
+	var allErr field.ErrorList
+
+	if m.Spec.Version == "" {
+		allErr = append(allErr, field.Invalid(field.NewPath("spec").Child("version"),
+			m.Name,
+			"spec.version' is missing"))
+	} else {
+		err := mssqlValidateVersion(m)
+		if err != nil {
+			allErr = append(allErr, field.Invalid(field.NewPath("spec").Child("version"),
+				m.Name,
+				err.Error()))
+		}
+	}
+
+	if m.IsStandalone() {
+		if ptr.Deref(m.Spec.Replicas, 0) != 1 {
+			allErr = append(allErr, field.Invalid(field.NewPath("spec").Child("replicas"),
+				m.Name,
+				"number of replicas for standalone must be one "))
+		}
+	} else {
+		if m.Spec.Topology.Mode == nil {
+			allErr = append(allErr, field.Invalid(field.NewPath("spec").Child("topology").Child("mode"),
+				m.Name,
+				".spec.topology.mode can't be empty in cluster mode"))
+		}
```

Comment:
How about defaulting the mode ?

Reply:
The default mode is Standalone now. (if spce.topology == nil)